nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/GetMongo.java

```diff
@@ -130,6 +138,31 @@ public ValidationResult validate(final String subject, final String value, final
         return propertyDescriptors;
     }
 
+    private ObjectMapper mapper = new ObjectMapper();
+
+    //Turn a list of Mongo result documents into a String representation of a JSON array
+    private String buildBatch(List<Document> documents) throws IOException {
+        List<Map> docs = new ArrayList<>();
+        for (Document document : documents) {
+            String asJson = document.toJson();
+            docs.add(mapper.readValue(asJson, Map.class));
+        }
+
+        return mapper.writeValueAsString(docs);
+    }
+
+    private void writeBatch(String payload, ProcessContext context, ProcessSession session) {
+        FlowFile flowFile = session.create();
+        flowFile = session.write(flowFile, new OutputStreamCallback() {
+            @Override
+            public void process(OutputStream out) throws IOException {
+                out.write(payload.getBytes("UTF-8"));
+            }
+        });
+        session.getProvenanceReporter().receive(flowFile, context.getProperty(URI).getValue());
```

Comment:
Could you also add the following?
````java
flowFile = session.putAttribute(flowFile, CoreAttributes.MIME_TYPE.key(), "application/json");
````
It's not mandatory but will improve UX when viewing flow file content in the UI.